check_dataset_functions.qmd

```diff
@@ -0,0 +1,267 @@
+
+## Functions to summarise excluded data
+
+### dataset_check_categorical_substitutions
+
+**output**: Table of categorical trait values that require substitutions
+
+If a categorical trait value is not identical to an allowed value in the corresponding trait concept in the accompanying trait dictionary (`traits.yml` file), it is moved to the `excluded_data` table. The following function generates a table of values requiring substitutions, as described in [adding data](adding_data_long.html#add_substitutions)
+
+The output table needs to be edited, to map an appropriate replacement for each unsupported trait value. This is generally most easily accomplish in Excel or a text editor.
+
+```
+dataset_check_categorical_substitutions <- function(database, dataset) {
+
+  required_substitutions <-
+    database$excluded_data %>%
+      dplyr::filter(
+        dataset_id == dataset,
+        error == "Unsupported trait value"
+      ) %>%
+      dplyr::distinct(dataset_id, trait_name, value) %>%
+      dplyr::rename(find = value) %>%
+      dplyr::select(-dataset_id) %>%
+      dplyr::mutate(replace = NA)
+  
+  required_substitutions
+}
+```
+
+### dataset_check_numeric_values
+
+**output**: Table of out-of-range numeric trait values:
+
+For numeric traits, if a trait value falls outside the allowed range, as defined for the specific trait concept in the accompanying trait dictionary (`traits.yml` file), it is moved to the `excluded_data` table. The following function generates a table of values that have been excluded. If the table is long, it is almost certainly due to a units-conversion error. This is not likely to be a list to save or edit, but simply confirming there isn't an error in how a trait was mapped or calculated and that the excluded trait values are legitimately excluded.
+
+```
+dataset_check_numeric_values <- function(database, dataset) {
+
+  out_of_range_values <-
+    database$excluded_data %>%
+      dplyr::filter(
+        dataset_id == dataset,
+        error == "Value out of allowable range"
+      ) %>%
+      dplyr::select(dataset_id, trait_name, value)
+  
+  out_of_range_values
+}
+```
+
+### dataset_check_taxonomic_updates
+
+**output**: Table of taxon names requiring taxonomic_updates
+
+An `aligned_name` in the `taxonomic_updates` table of a newly added dataset might not be in the database's `taxon_list.csv` file for two reasons: 1) It requires aligning due to typos, non-standard syntax; 2) The `taxon_list.csv` file does not include all possible taxon names and needs to be updated from an external resource. Each database requires its own taxonomy functions and taxonomic references, but this function creates a list of names that require further effort to align.
+
+```
+dataset_check_taxonomic_updates <- function(taxon_list, database, dataset) {
+
+  taxon_list <- readr::read_csv("config/taxon_list.csv")
+  
+  names_to_align <- 
+    database$taxonomic_updates %>%
+      dplyr::filter(dataset_id == current_study) %>%
```

Comment:
`current_study` has not been defined, should be `dataset`?

Reply:
yes! I wasn't being careful when I copied my code.